modules/l4proxyprotocol/matcher.go

```diff
@@ -46,39 +44,15 @@ func (MatchProxyProtocol) CaddyModule() caddy.ModuleInfo {
 
 // Match returns true if the connection looks like it is using the Proxy Protocol.
 func (m MatchProxyProtocol) Match(cx *layer4.Connection) (bool, error) {
-	p := make([]byte, 4)
-	_, err := io.ReadFull(cx, p)
+	buf := make([]byte, 12) // len(headerV2Prefix)
```

Comment:
Maybe directly using `len(headerV2Prefix)` is better? If we adjust the header bytes in the future, although seems impossible, then there's no need to update the number again.

Reply:
This was premature optimization from me 😄   
Since headerV2Prefix is not a const I think the compiler can not precalculate the length, meaning it is executed on every request.  
The performance difference is probably not measurable, and I also did not verify if this is actually true.  
I changed it.